test/support/pages/Common.js

```diff
@@ -0,0 +1,79 @@
+// in test/support/pages/Common.js
+define(function (require) {
+
+  var registerSuite = require('intern!object');
+  var expect = require('intern/dojo/node!expect.js');
+  var Promise = require('bluebird');
+  var moment = require('moment');
+
+
+  // the page object is created as a constructor
+  // so we can provide the remote Command object
+  // at runtime
+  function Common(remote) {
+    this.remote = remote;
+  }
+
+  var defaultTimeout = 5000;
+
+  Common.prototype = {
+    constructor: Common,
+
+
+    tryForTime: function tryForTime(timeout, block) {
+      var self = this;
+      var start = Date.now();
+      var lastTry = 0;
+
+      function attempt() {
+        lastTry = Date.now();
+
+        if (lastTry - start > timeout) {
+          throw new Error('timeout');
+        }
+
+        return Promise
+          .try(block)
+          .then(function tryForTimeSuccess() {
+            self.log('tryForTime success in about ' + (lastTry - start) + ' milliseconds');
+            return (lastTry - start);
+          })
+          .catch(function tryForTimeCatch(err) {
+            self.log('failed with "' + err.message + '"');
+            self.log('trying again in 1/2 second');
+            return Promise.delay(500).then(attempt);
+          });
+      }
+
+      return Promise.try(attempt);
+    },
+
+
+    log: function log(logString) {
+      console.log(moment().format('HH:mm:ss.SSS') + ': ' + logString);
+    },
+
+
+    sleep: function sleep(sleepMilliseconds) {
+      var self = this;
```

Comment:
could this be a promise that resolves after sleepMilliseconds instead of looking for an element?